File: app/src/main/java/com/sabihamumcu/kelimekutusu/viewmodel/WordViewModel.java
package com.sabihamumcu.kelimekutusu.viewmodel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.sabihamumcu.kelimekutusu.model.Word;
import com.sabihamumcu.kelimekutusu.repo.WordRepository;

import java.util.List;

public class WordViewModel extends AndroidViewModel {

    private WordRepository wordRepository;
    private LiveData<List<Word>> allWords;

    public WordViewModel(@NonNull Application application) {
        super(application);
        wordRepository = new WordRepository(application);
        allWords=wordRepository.getWords();
    }

    public void insertWord(Word word) {
        wordRepository.insertWord(word);
    }

    public void updateWord(Word word) {
        wordRepository.update(word);
    }

    /*public LiveData<Word> getWord(String id) {
        return wordRepository(id);
    }*/

    public LiveData<List<Word>> getWords() {
        return wordRepository.getWords();
    }

    public void deleteWord(Word word){
        wordRepository.delete(word);
    }

    public Word findWord(String word){
        return wordRepository.find(word);
    }

}

File: app/src/main/java/com/sabihamumcu/kelimekutusu/viewmodel/WordViewModel.kt
package com.sabihamumcu.kelimekutusu.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.sabihamumcu.kelimekutusu.model.Word
import com.sabihamumcu.kelimekutusu.repo.WordRepository

class WordViewModel(application: Application) : AndroidViewModel(application) {

    private val wordRepository: WordRepository
    private val allWords: LiveData<List<Word?>?>?

    fun insertWord(word: Word?) {
        wordRepository.insertWord(word)
    }

    fun updateWord(word: Word?) {
        wordRepository.update(word)
    }

    val words: LiveData<List<Word?>?>?
        get() = wordRepository.words

    fun deleteWord(word: Word?) {
        wordRepository.delete(word)
    }

    fun findWord(word: String?): Word {
        return wordRepository.find(word)
    }

    init {
        wordRepository = WordRepository(application)
        allWords = wordRepository.words
    }
}